dask_geopandas/geohash.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import pandas as pd
+from pygeohash import encode
+
+
+def _geohash(gdf, precision):
+
+    """
+    Calculate geohash for the mid points of each geometry
+    int coordinates
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+    precision : int
+        precision of the Geohash
+
+    Returns
+    ---------
+    Pandas Series containing geohash
+    """
+
+    # Calculate bounds
+    bounds = gdf.bounds.to_numpy()
+    # Calculate mid points based on bounds
+    x_mids, y_mids = _calculate_mid_points(bounds)
+    # Vectorize geohash for fast speed up
+    geohash_vec = np.vectorize(encode)
+    # Encode mid points of geometries using geohash
+    geohash = geohash_vec(y_mids, x_mids, precision)
+
+    return pd.Series(geohash, index=gdf.index, name="geohash")
+
+
+def _calculate_mid_points(bounds):
```

Comment:
This is used elsewhere as well. Can you centralise these util functions and reuse the same one?

Reply:
yes, I was thinking this function could also be centralized alongside the other functions that have been previously discussed. Something like spatial_partition.py? 

Would it be wise to create a separate PR request to centralize several of these functions?

Reply:
> yes, I was thinking this function could also be centralized alongside the other functions that have been previously discussed. Something like spatial_partition.py?

Yeah, something like that. maybe with `utils` in that so we know what it is for.

> Would it be wise to create a separate PR request to centralize several of these functions? 

Yeah, once this is merged we can move it around.

Reply:
Updated in 9947e80